### PR TITLE
Travis CI makes archives and uploads them GitHub releases

### PR DESCRIPTION
**Pull request by jkawamoto**
When a tag is pushed to GitHub, Travis CI will make zip files including the executable jar for the current version and dependent libraries, and then upload them to GitHub releases.

### CODE REVIEW

**Comment by kaloyan-raev** [comment]
Thanks for the effort to assemble this, but I am not completely sure about this PR.

I am not a fan of hosting binaries of dependencies in the source code repository. Perhaps, I should work with the storj team to include the dlls and dylibs in the releases of the libstorj project. Then we can make Travis CI to download and unpack them from there.

The greater issue is with the release flow. Usually I declare a release in the project when I have successfully tested the app on Windows. So, I have already the windows package assembled manually. Having it automatically built on release tag isn't really helpful. If I first release and only then test, what happens if the test fails (because of something small that I forgot to update)? Fix and make another release?

This whole thing would be really helpful if we have it for snapshots too. Once we verified that everything is correct with the latest release candidate snapshot, we promote it to a release.

File: vendor/win/goobox-sync-storj.bat [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+@rem Set local scope for the variables with windows NT shell
+if "%OS%"=="Windows_NT" setlocal
+
+set DIRNAME=%~dp0
+if "%DIRNAME%" == "" set DIRNAME=.
+set APP_BASE_NAME=%~n0
+set APP_HOME=%DIRNAME%
+
+@rem goobox-sync-storj app version
+set GOOBOX_SYNC_STORJ_VERSION=0.0.21
```
Comment (kaloyan-raev) [inline vendor/win/goobox-sync-storj.bat]: This version number should not be hardcoded. It should be a variable/property that the build replaces with the actual version number of the release.

**Comment by jkawamoto** [comment]
About the dlls, I'm fine with anything if it reduces manual tasks. The Storj team is working for including dynamic libraries to their releases https://github.com/Storj/libstorj/issues/228. Following up on that issue would be a good idea. 

The binaries built on the CI service are mainly for including in the GUI releases. The goal of automating this is that binaries for both Windows and Mac (and Linux in the future) can be automatically generated and uploaded just by tagging and pushing. This doesn't change our release flow except that you won't need to upload the Windows binary and I won't need to manually build and upload the Mac version.

Btw, if you want to post release candidates to Slack as the GUI does, I can implement that too.

File: vendor/win/goobox-sync-storj.bat (hunk 1 above)
Comment (jkawamoto) [reply vendor/win/goobox-sync-storj.bat]: I'm using [bumpversion](https://github.com/peritus/bumpversion) to manage version numbers. I think it is easier than replacing version numbers by `sed` or something like that in the CI server. 

Actually, the .bat file is made by Gradle for sync-sia. So, perhaps Maven has a similar plugin to make .bat for each release.

**Comment by kaloyan-raev** [comment]
I uploaded the windows dlls in the java-libstorj release: https://github.com/Storj/java-libstorj/releases/tag/0.7.3. It includes not only the libstorj dll and its dependent dlls, but also the storj-java.dll with the JNI native code.

This is going to work, right? Let me know if the current format is OK. I can change it as required. We can also upload the mac dylib binaries the same way. I will also think about automating those to be prepared on each java-libstorj release.

Regarding the dlls for overlay icons and context menu, we can upload them to the github repo. They are small and not supposed to change often, if at all. And I cannot think about a better place for them.

File: vendor/win/goobox-sync-storj.bat (hunk 1 above)
Comment (kaloyan-raev) [reply vendor/win/goobox-sync-storj.bat]: I am OK moving to Gradle.

**Comment by jkawamoto** [comment]
The dll files in https://github.com/Storj/java-libstorj/releases/tag/0.7.3 looks good. If dylibs can be provided there too, that'd be good.

**Comment by jkawamoto** [comment]
I modified `.travis.yml` so that it will download dlls and dylibs from java-libstorj's release page, and remove those libraries from this repository.

**Pull request merged**